services/github/github-pull-request-check-state.service.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const Joi = require('joi')
+const countBy = require('lodash.countby')
+const GithubAuthService = require('./github-auth-service')
+const { fetchIssue } = require('./github-common-fetch')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.object({
+  state: Joi.equal('failure', 'pending', 'success').required(),
+  statuses: Joi.array()
+    .items(
+      Joi.object({
+        state: Joi.equal('error', 'failure', 'pending', 'success').required(),
+      })
+    )
+    .default([]),
+}).required()
+
+module.exports = class GithubPullRequestCheckState extends GithubAuthService {
+  static get category() {
+    return 'other'
+  }
+
+  static get url() {
+    return {
+      base: 'github/status',
+      format: '(s|contexts)/pulls/([^/]+)/([^/]+)/(\\d+)',
+      capture: ['which', 'user', 'repo', 'number'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub pull request check state',
+        urlPattern: 's/pulls/:user/:repo/:number',
+        staticExample: this.render({ which: 's', state: 'pending' }),
+        exampleUrl: 's/pulls/badges/shields/1110',
+        keywords: ['GitHub', 'pullrequest', 'detail', 'check'],
+        documentation,
+      },
+      {
+        title: 'GitHub pull request check contexts',
+        urlPattern: 'contexts/pulls/:user/:repo/:number',
+        staticExample: this.render({
+          which: 'contexts',
+          stateCounts: { passed: 5, pending: 1 },
```

Comment:
this example should also be passing the summary state param as well as the raw state counts

Reply:
Do you mean, adding in `state` even though it's unused? I'm okay with that.